### ISSUE TITLE
[Xcode 16] Asset validation failed issue when distributing app to TestFlight

### ISSUE THREAD

**Issue by luizhfff** [star-micronics/react-native-star-io10#126]
## Description
After Xcode was updated to v16, started to get this issue with asset validation when distributing iOS app to TestFlight.

This is a React Native project

I have ENABLE_BITCODE = NO configured in the build settings and we haven't changed anything in the project regarding starIO package.

Has anyone been able to find a workaround for this issue?

Thanks

## Your device where the bug occurs 
 - NA

## Your printer
- NA

## Your development environment
ProductName:		macOS
ProductVersion:		15.0
BuildVersion:		24A335

## Additional context
![image](https://github.com/user-attachments/assets/c772b444-5360-4599-93a9-2f2780b0b1f6)
![image](https://github.com/user-attachments/assets/0088b6bf-c2bc-4bd1-a426-53718ca1f179)

**Comment by luizhfff** [comment]
I was actually able to work around this issue by adding this piece of code to Podfile:

After `post_install do |installer| block` :

```
# Removing bitcode from the framework
  bitcode_strip_path = `xcrun --find bitcode_strip`.chop!

  def strip_bitcode_from_framework(bitcode_strip_path, framework_relative_path)
    framework_path = File.join(Dir.pwd, framework_relative_path)
    command = "#{bitcode_strip_path} #{framework_path} -r -o #{framework_path}"
    puts "Stripping bitcode: #{command}"
    system(command)
  end

  framework_paths = [
    "../node_modules/react-native-star-io10/ios/libs/StarIO10.xcframework/ios-arm64/StarIO10.framework/StarIO10",
    "../node_modules/react-native-star-io10/ios/libs/StarIO10.xcframework/ios-arm64_x86_64-simulator/StarIO10.framework/StarIO10",
  ]

  framework_paths.each do |framework_relative_path|
    strip_bitcode_from_framework(bitcode_strip_path, framework_relative_path)
  end
```

Then, in the iOS folder, removed Pods folder (`pod deintegrate && pod cache clean --all`) and ran `pod install --repo-update`

**Comment by luizhfff** [comment]
BIG WARNING:
I was able to distribute the app after stripping bitcode with code above, but now I am not able to find printers...
https://github.com/star-micronics/react-native-star-io10/issues/124

**Comment by bandit-ibayashi** [comment]
@luizhfff Thank you for posting your issue.
Could you please confirm and let us know your `react-native-star-io10` version?

We had updated it not to contain the bitcode at version [1.3.0](https://github.com/star-micronics/react-native-star-io10/releases/tag/V1.3.0).
So it would be very helpful if you could first tell us what version of the sample you are using.

According to the Apple engineer's response on the Apple forum, we understand that this error occurs when using libraries that contain Bitcode.
https://forums.developer.apple.com/forums/thread/763562?answerId=805557022#805557022
Also, checking the Issue postings for other libraries, it seems that Bitcode checks have been made stricter than in Xcode16.

**Comment by luizhfff** [comment]
> @luizhfff Thank you for posting your issue. Could you please confirm and let us know your `react-native-star-io10` version?
> 
> We had updated it not to contain the bitcode at version [1.3.0](https://github.com/star-micronics/react-native-star-io10/releases/tag/V1.3.0). So it would be very helpful if you could first tell us what version of the sample you are using.
> 
> According to the Apple engineer's response on the Apple forum, we understand that this error occurs when using libraries that contain Bitcode. https://forums.developer.apple.com/forums/thread/763562?answerId=805557022#805557022 Also, checking the Issue postings for other libraries, it seems that Bitcode checks have been made stricter than in Xcode16.

hey @bandit-ibayashi , thanks for replying...

We were using version 1.2.0 but I also tried to update to the latest version 1.7.0. But after updating it the app could not find printers anymore...

I will try version 1.3.0 like you suggested

**Comment by bandit-ibayashi** [comment]
@luizhfff Thank you for letting us know the version you are using.
If the BITCODE issue no longer occurs in 1.7.0 without the interim workaround, then this one is likely resolved in the upgrade.

And we believe that the problem of not being able to find the printer is a separate issue from this. As you have already noticed, if you are using the Ethernet (LAN) printers, the missing printers may be related to this issue (#124) .
This issue is scheduled to be fixed in early November.

Thank you for your patience.

**Comment by bandit-ibayashi** [comment]
@luizhfff I am sorry for my late notice. We had released version [2.8.0](https://github.com/star-micronics/react-native-star-io10/releases/tag/1.8.0), which includes a fix for this issue. Thank you for your patience.
